app/Http/Transformers/AssetsTransformer.php

```diff
@@ -64,6 +68,8 @@ public function transformAsset(Asset $asset)
                 'name'=> e($asset->defaultLoc->name)
             ]  : null,
             'image' => ($asset->getImageUrl()) ? $asset->getImageUrl() : null,
+            'qr' => ($setting->qr_code=='1') ? config('app.url').'/uploads/barcodes/qr-'.str_slug($asset->asset_tag).'-'.str_slug($asset->id).'.png' : null,
```

Comment:
If this works then great, but I don't like concatenating `config('app.url')` to various bits of string, and then running `str_slug($asset->blah)` in this part. This feels like some kind of `route()` facade should be able to generate this easily enough. And if we are using or re-using a thing like this frequently, and it doesn't fit in a route, then maybe in a Helper or something?

And similar with the 'alt barcode' below.

Regardless of some misgivings, I'll accept - this will probably be useful to the people who need it. I just don't like duplicating URL-assembling logic in several places throughout the code.

Reply:
We only use that in two places so it didn't seem worth writing a helper for it.

Reply:
(That said I wouldn't kick a re-factor out of bed down the line if people end up using this.)